### PR TITLE
Fix interface generation if @inline is present

### LINKED ISSUE #425: Interface generation does not work if @inline is present
E.g. if I have a file `Example.res`:

```rescript
@inline
let gray = "#f0f0f0"
```

the create-interface command yields me this broken `Example.resi` file:

```rescript
@inline
let gray = "#f0f0f0"
```

expected:

```rescript
@inline
let gray: string
```

### PR DESCRIPTION
**Pull request by An-Tu**
Hello! I'm familiar with ReScript but it's my first try with OCaml. So if I did something wrong please feel free to correct me.

fix #425

### CODE REVIEW

**Comment by zth** [comment]
Awesome! 🎉 I'll let @cristianoc comment on the OCaml itself, but the output generated by the test looks great.

**Comment by An-Tu** [comment]
I realized that my new function `SourceFileExtractor.extractLine` isn't needed. So I removed it.

File: analysis/src/CreateInterface.ml [hunk 1]
```diff
@@ -162,15 +162,39 @@ let printSignature ~extractor ~signature =
...
+        | [] -> ""
+        | firstLine :: _ -> firstLine
+      in
+      let inlineAttr = "@inline" in
+      let inlineAttrLen = String.length inlineAttr in
+      let attrStartIdx = String.index firstLine '@' in
+
+      if
+        try String.sub firstLine attrStartIdx inlineAttrLen = inlineAttr
+        with _ -> false
```
Comment (cristianoc) [inline analysis/src/CreateInterface.ml:183]: One would catch the specific exception `| Invalid_argument _ ->` .

But here one can also check the size beforehand.

**Review by cristianoc — COMMENTED** [review]
Can you add 2 examples to the test set:
```rescript
@genType
@inline
let f2 = 10

@genType @inline
let f3 = 10
```

see what happens

**Comment by cristianoc** [comment]
> Hello! I'm familiar with ReScript but it's my first try with OCaml. So if I did something wrong please feel free to correct me.
> 
> fix #425

That's an impressive first try!

**Comment by An-Tu** [comment]
> Can you add 2 examples to the test set:
> 
> ```rescript
> @genType
> @inline
> let f2 = 10
> 
> @genType @inline
> let f3 = 10
> ```
> 
> see what happens

I missed that there can be more than one attribute. Fixed...I hope :)

File: analysis/src/CreateInterface.ml [hunk 2]
```diff
@@ -31,6 +31,58 @@ module SourceFileExtractor = struct
...
+           if trimmedLine |> String.length > 0 && String.get trimmedLine 0 = '@'
+           then (
+             let attrDecIdxSeparator = ref (-1) in
+             let i = ref 1 in
+             let lineLength = String.length line in
+
+             while !attrDecIdxSeparator = -1 do
+               if !i < lineLength then
+                 let idx = !i in
+                 let currentChar = line.[idx] in
```
Comment (cristianoc) [inline analysis/src/CreateInterface.ml:59]: How do we know that `line[1]` exists?

Comment (cristianoc) [reply analysis/src/CreateInterface.ml:59]: I think it can be made more obvious.

Comment (An-Tu) [reply analysis/src/CreateInterface.ml:59]: Oh, thanks. I think I get it. I'll do it.

**Comment by An-Tu** [comment]
@cristianoc One more try :)

File: analysis/src/CreateInterface.ml [hunk 3]
```diff
@@ -31,6 +31,82 @@ module SourceFileExtractor = struct
...
+
+  let getLineIdx ((lineIdx, _), _) = lineIdx
+
+  let getAttrOffset ((_, attrOffset), _) = attrOffset
+
+  let getAttrName (_, attribute) = attribute
+
+  let makeAttr lineIdx attrOffsetStart attrOffsetEnd line =
+    ( (lineIdx, attrOffsetStart),
+      String.sub line attrOffsetStart (attrOffsetEnd - attrOffsetStart) )
```
Comment (cristianoc) [inline analysis/src/CreateInterface.ml:55]: Nit: to know that this is safe (not out of bound) you need to look at the callers of this function.
But if this function were to be moved into `make`, which is the only user, then the evidence that this is safe would be right there.

File: analysis/src/CreateInterface.ml [hunk 4]
```diff
@@ -31,6 +31,82 @@ module SourceFileExtractor = struct
...
+           done;
+
+           match !state with
+           | Collect attrOffset ->
+             res :=
+               makeAttr lineIdx attrOffset (String.length line) line :: !res
+           | _ -> ());
+    !res |> List.rev
+
+  let rec contain attributeForSearch t =
```
Comment (cristianoc) [inline analysis/src/CreateInterface.ml:79]: This is OK.
It could also be a one-line `List.find`.

File: analysis/src/CreateInterface.ml [hunk 5]
```diff
@@ -162,15 +238,29 @@ let printSignature ~extractor ~signature =
...
-         Copy the external declaration verbatim from the implementation file *)
+      (* Rescript primitive name, e.g. @val external ... *)
       let lines =
         let posStart, posEnd = Loc.range val_loc in
         extractor |> SourceFileExtractor.extract ~posStart ~posEnd
       in
-      Buffer.add_string buf ((lines |> String.concat "\n") ^ "\n");
+      let attributes = AttributesUtils.make lines in
+
+      if AttributesUtils.contain "@inline" attributes then
```
Comment (cristianoc) [inline analysis/src/CreateInterface.ml:250]: This is nice and clean.

**Review by cristianoc — APPROVED** [review]
This is looking great!
Let some style comments/thoughts. Take / ignore any one.

**Comment by An-Tu** [comment]
@cristianoc Thanks a lot for your review! I fixed your comments.

**Pull request merged**